src/main/java/com/commercetools/pspadapter/tenant/TenantProperties.java

```diff
@@ -0,0 +1,138 @@
+package com.commercetools.pspadapter.tenant;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.validation.annotation.Validated;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import java.util.List;
+
+@Configuration
+@ConfigurationProperties("tenantConfig")
+@Validated
+public class TenantProperties {
+
+    @Valid
+    private List<Tenant> tenants;
+
+    public List<Tenant> getTenants() {
```

Comment:
shy not a HashMap? than you do:
```java
tenantProperties.getTenants().stream()
    .filter(t -> tenantName.equals(t.getName()))
    .findAny();
```
what is super inefficient in this case!

Basically, you should not expose the tenants collection at all, add method: `Optional<Tenant> TenantProperties#getTenantByName(String name)`